app/src/main/java/com/example/plantsapp/presentation/ui/worker/ReschedulingWorker.kt

```diff
@@ -0,0 +1,41 @@
+package com.example.plantsapp.presentation.ui.worker
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.example.plantsapp.domain.repository.PlantsRepository
+import com.example.plantsapp.domain.repository.TasksRepository
+import com.example.plantsapp.domain.usecase.GetTasksForPlantAndDateUseCase
+import com.example.plantsapp.presentation.ui.utils.minusDays
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import java.util.Date
+
+@HiltWorker
+class ReschedulingWorker @AssistedInject constructor(
+    @Assisted private val context: Context,
+    @Assisted workerParams: WorkerParameters,
+    private val plantsRepository: PlantsRepository,
+    private val getTasksForPlantAndDateUseCase: GetTasksForPlantAndDateUseCase,
+    private val tasksRepository: TasksRepository
+) : CoroutineWorker(context, workerParams) {
+
+    override suspend fun doWork(): Result {
+
+        plantsRepository.fetchPlants()
+            .map { plant ->
```

Comment:
Also: use map when you need to return something, in other case use forEach

For example in this case you use map but do not use result of this map